### ISSUE TITLE
Select strange behaviour

### ISSUE THREAD

**Issue by AREXP** [scylladb/scylla#2272]
*Installation details*
Scylla version (or git commit hash): 1.6.3-20170405.5ab8601
Cluster size: 1
OS : Debian 8.7

Hello, I have such a table:
```
CREATE TABLE dramiel_test.hour_data (                                                                                                
    bucket text,                                                                                                                     
    hour_ts int,                                                                                                                     
    ug int,                                                                                                                          
    user bigint,                                                                                                                     
    ad_clk set<int>,                                                                                                                 
    ad_exp set<int>,                                                                                                                 
    segments set<int>,                                                                                                               
    site_clk set<int>,                                                                                                               
    site_exp set<int>,                                                                                                               
    PRIMARY KEY (bucket, hour_ts, ug, user)                                                                                          
) WITH CLUSTERING ORDER BY (hour_ts ASC, ug ASC, user ASC)                                                                           
    AND bloom_filter_fp_chance = 0.01                                                                                                
    AND caching = '{"keys":"ALL","rows_per_partition":"ALL"}'                                                                        
    AND comment = ''
    AND compaction = {'class': 'LeveledCompactionStrategy'}
    AND compression = {'sstable_compression': 'org.apache.cassandra.io.compress.LZ4Compressor'}
    AND dclocal_read_repair_chance = 0.1
    AND default_time_to_live = 0
    AND gc_grace_seconds = 864000
    AND max_index_interval = 2048
    AND memtable_flush_period_in_ms = 0
    AND min_index_interval = 128
    AND read_repair_chance = 0.0
    AND speculative_retry = '99.0PERCENTILE';
```
Now I'm testing the table and putting there a piece of data(about 1/48 size of real data). 
When i try to select some data I get some strange output.
For example, when I do the next query:
```
sql = """
SELECT hour_ts, ug, user, segments, site_exp, site_clk, ad_exp, ad_clk
FROM hour_data
WHERE bucket = '2017-29-03'
"""
counted = defaultdict(int)
res = session.execute(sql)
for num, row in enumerate(res):
    counted[(row["hour_ts"], row["ug"])] += 1

for (hour, ug), count in sorted(counted.iteritems()):
    print DateTime.from_timestamp_naive(hour), ug, count
```
Output is. Show only one hour/hour_ts:
```
2017-03-29 00:00:00 1 9969
2017-03-29 00:00:00 2 10043
2017-03-29 00:00:00 3 10032
2017-03-29 00:00:00 4 9871
2017-03-29 00:00:00 5 9830
2017-03-29 00:00:00 6 10028
2017-03-29 00:00:00 7 10024
2017-03-29 00:00:00 8 9901
2017-03-29 00:00:00 9 9845
2017-03-29 00:00:00 10 10103
2017-03-29 00:00:00 11 10144
2017-03-29 00:00:00 12 9934
2017-03-29 00:00:00 13 9732
2017-03-29 00:00:00 14 10000
2017-03-29 00:00:00 15 9860
2017-03-29 00:00:00 16 9860
...cut output for convenience...
```
Output above is totally correct, but when I'm trying filter explicitly by hour_ts and ug:
```
sql = """
SELECT hour_ts, ug, user, segments, site_exp, site_clk, ad_exp, ad_clk
FROM hour_data
WHERE bucket = '2017-29-03' AND hour_ts = %s AND ug = %s
"""

for hour in period.iter_hours():
    # Iterating by hour_ts
    hour_ts = hour.timestamp()
    counted = defaultdict(int)
    for ug in range(1, 17):
        res = session.execute(sql, (hour_ts, ug))
        for num, row in enumerate(res):
            counted[(row["hour_ts"], row["ug"])] += 1

    for (hour, ug), count in sorted(counted.iteritems()):
        print DateTime.from_timestamp_naive(hour), ug, count
```
I get:
```
2017-03-29 00:00:00 2 3
2017-03-29 00:00:00 3 68
2017-03-29 00:00:00 4 10
2017-03-29 00:00:00 5 38
2017-03-29 00:00:00 6 9
2017-03-29 00:00:00 7 5
2017-03-29 00:00:00 8 17
2017-03-29 00:00:00 9 45
2017-03-29 00:00:00 10 26
2017-03-29 00:00:00 11 53
2017-03-29 00:00:00 12 33
2017-03-29 00:00:00 13 28
2017-03-29 00:00:00 14 25
2017-03-29 00:00:00 15 17
2017-03-29 00:00:00 16 11
...cut output for convenience...
```
Output above totally incorrect, looks like data is missed, but from first example you can see that data is OK. Also I noticed one strange thing: when the table almost empty and there is only one hour data, then explicit filtering by this hour is correct, but after a while, when table have data for more hour the output is messed up.
Am I doing anything wrong?

[Full correct output](https://pastebin.com/6WxtRyYd)
[Full incorrect output](https://pastebin.com/4FBnRJVQ)

**Comment by tgrabiec** [comment]
@AREXP I was not able to reproduce with my data set. Does the problem persist after you restart the database? Can you upload your sstables to an S3 bucket? Instructions are here: https://github.com/scylladb/scylla/wiki/How-to-report-a-Scylla-problem#send-the-core-dump

**Comment by AREXP** [comment]
Sorry for waiting. I have some troubles with getting a coredump.
As I read from the given instruction, coredump is creating when Scylla fails, falls, crashing etc. But my installation is a quite stable and I couldn't get any coredump.
Is there more "explicit" way to generate a coredump?

**Comment by gleb-cloudius** [comment]
On Mon, Apr 17, 2017 at 11:45:46PM -0700, AREXP wrote:
> Is there more "explicit" way to generate a coredump?
> 
kill -ABRT $scylla_pid

--
			Gleb.

**Comment by tgrabiec** [comment]
@AREXP I think I confused you. The link was just to show how to upload files to an S3 bucket. I didn't ask for a core dump. I need the data files for your table (dramiel_test.hour_data), which are by default located on disk in `/var/lib/scylla/data/`.

**Comment by AREXP** [comment]
Yes, you did=)
Uploaded two files under uuid `adc9ed62-5ee5-4343-93b5-9a7171d773e9`

**Comment by tgrabiec** [comment]
@slivne Probably introduced in 1.4 by 700feda0db9aa781635758f5b73487d46a692dab

**Comment by AREXP** [comment]
When you gonna release build with bugfix?

**Comment by tzach** [comment]
@AREXP if all goes well, in a day

**Comment by slivne** [comment]
fixed in 1.6.4
